packages/mui-material/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -166,7 +166,8 @@ const SwipeableDrawer = React.forwardRef(function SwipeableDrawer(inProps, ref)
 
   const swipeAreaRef = React.useRef();
   const backdropRef = React.useRef();
-  const paperRef = React.useRef();
+  const localPaperRef = React.useRef();
+  const paperRef = PaperProps.ref || localPaperRef;
```

Comment:
Have you considered using `useForkRef`? This means that both the local paper ref and paperRef would have the dom node attached to `current`. There are instances of this usage in other components for reference.

Reply:
@mnajdova i just checked how useForkRef is implemented and used, you are right it makes sense to use useForkRef in this case and i updated code accordingly to use it.